### ISSUE TITLE
BPDU transmission period

### ISSUE THREAD

**Issue by atalaska** [adigostin/mstp-lib#11]
In other implementations of RSTP that I've seen, the BPDU is only transmitted every hello time, or when there is a topology change:  1 BPDU per hello period in stable conditions.
Toying with the simulator, I have noticed that, on every hello period, 2 BPDUs are transmitted from the designated ports. Looks like one of the BPDUs received on the root port triggers the transmission of a BPDU on the designated ports, which is added to  the BPDU transmitted on hello time.

This seems like unnecessary traffic. Is this intentional? Does this respect the 802.1 standard?
Thanks

**Comment by adigostin** [comment]
Would it be possible to list the repro steps? I don't think I ever saw this.. Two BPDUs per HelloTime seems serious enough and such behavior wouldn't have escaped me. Thanks.

**Comment by atalaska** [comment]
Thanks for your prompt response.
See files attached, as you can see, the 3rd bridge receives 2 BPDU every 2 seconds (hello time).
[doubleBPDU.stp.zip](https://github.com/adigostin/mstp-lib/files/2660191/doubleBPDU.stp.zip)
![2018-12-08 18_30_16-doublebpdu stp](https://user-images.githubusercontent.com/45670958/49692772-7de85d80-fb17-11e8-868b-8e86aa09437d.png)

**Comment by adigostin** [comment]
Ok, it's pretty clear. Thanks for providing the repro. I did try the configuration with three switches before my first answer; I looked at the tranmitted BPDUs only, which seemed to be in order. I'll look into this bug and I'll fix it sometime next week..

**Comment by adigostin** [comment]
I fixed this in the develop branch. See https://github.com/adigostin/mstp-lib/commit/8fe62bd63d2c312b0a55a972ff7be449fb36fef1

The bug was in my implementation of paragraph j) from page 378 of 802.1Q-2011: 
> Additionally, updtInfo is set if the port priority vector differs from the designated priority vector or the port’s associated timer parameter(s) differ(s) from the Root Port’s associated timer parameters
My understanding of this paragraph when I initially wrote the code was that "associated timer parameters" means `portTimes`. Turns out it actually means `designatedTimes` (this is consistent with the first half of the condition, the one about priority vectors). I changed the code accordingly and now the behavior is the expected one: one BPDU sent per HelloTime.
